modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java

```diff
@@ -393,13 +392,30 @@ public void close() {
         versionChainTree.close();
     }
 
+    /**
+     * Removes all data from this storage and frees all associated resources.
+     *
+     * @throws StorageException If failed to destroy the data or storage is already stopped.
+     */
+    public void destroy() {
+        try {
+            versionChainTree.destroy();
```

Comment:
Why did you add this? This implementation makes no sense for persistent storage and is just wrong for volatile storage. And, it's synchronous. I don't like it.